library/src/main/java/com/google/android/exoplayer/MediaCodecTrackRenderer.java

```diff
@@ -1139,7 +1139,8 @@ private static boolean codecNeedsEosPropagationWorkaround(String name) {
    *     buffer with {@link MediaCodec#BUFFER_FLAG_END_OF_STREAM} set. False otherwise.
    */
   private static boolean codecNeedsEosFlushWorkaround(String name) {
-    return Util.SDK_INT <= 23 && "OMX.google.vorbis.decoder".equals(name);
+    return (Util.SDK_INT <= 23 && "OMX.google.vorbis.decoder".equals(name))
+        || (name != null && name.startsWith("OMX.amlogic.avc.decoder.awesome"));
```

Comment:
Unless we have evidence this affects OMX.amlogic.avc.decoder.awesome on *all* devices and API levels, this seems overly broad. Can we restrict the check to the specific device and a lower API level limit, unless we have evidence that it needs to be broader than that? For example:

return (Util.SDK_INT <= 23 && "OMX.google.vorbis.decoder".equals(name))
    || (Util.SDK_INT <= 19 && "OMX.amlogic.avc.decoder.awesome".equals(name) && Util.DEVICE.equals("xxx");

Where you need to replace xxx with whatever Util.DEVICE is equal to on the affected device. Thanks!